internal/executor/taskfilter/taskfilter.go

```diff
@@ -18,14 +18,64 @@ func MatchExactTask(desiredTaskName string) TaskFilter {
 		var filteredTasks []*api.Task
 
 		for _, task := range tasks {
-			if strings.EqualFold(desiredTaskName, task.Name) {
-				// Clear task's dependencies
-				task.RequiredGroups = task.RequiredGroups[:0]
+			// Ensure that this task's name matches with the name we're looking for
+			desiredTaskNameLower := strings.ToLower(desiredTaskName)
+			taskNameLower := strings.ToLower(task.Name)
 
-				filteredTasks = append(filteredTasks, task)
+			if !strings.HasPrefix(desiredTaskNameLower, taskNameLower) {
+				continue
 			}
+
+			// In case we're looking for a task with specific labels — extract them and ensure they all match
+			desiredLabels := extractLabels(strings.TrimPrefix(desiredTaskNameLower, taskNameLower))
+
+			var actualLabels []string
+			if task.Metadata != nil {
+				actualLabels = task.Metadata.UniqueLabels
+			}
+
+			if !containsAll(desiredLabels, actualLabels) {
+				continue
+			}
+
+			// Clear task's dependencies
+			task.RequiredGroups = task.RequiredGroups[:0]
+
+			filteredTasks = append(filteredTasks, task)
 		}
 
 		return filteredTasks
 	}
 }
+
+func extractLabels(s string) (result []string) {
+	labels := strings.Split(s, " ")
+
+	// Filter out empty labels
+	for _, label := range labels {
+		if strings.TrimSpace(label) == "" {
+			continue
+		}
+
+		result = append(result, label)
+	}
+
+	return
+}
+
+func matchLabels(soughtLabels, actualLabels []string) (numMatchedLabels int) {
+	for _, soughtLabel := range soughtLabels {
+		for _, actualLabel := range actualLabels {
+			if strings.EqualFold(soughtLabel, actualLabel) {
+				numMatchedLabels++
+				break
+			}
+		}
+	}
+
+	return
+}
+
+func containsAll(desiredLabels, actualLabels []string) bool {
+	return matchLabels(desiredLabels, actualLabels) == len(desiredLabels)
+}
```

Comment:
I meant something like:

```suggestion
func containsAll(values []string, items []string) bool {
	ItemsLoop:
	for _, item := range items {
		for _, value := range values {
			if strings.EqualFold(value, item) {
				continue ItemsLoop
			}
		}
		return false
	}
	return true
}
```

Reply:
I don't really like the continue label since it needlessly breaks the program's structure.

However I've moved everything into a single function in b6ce7b0.

Does that work for you?